src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/IncrementingPollingCounter.cs

```diff
@@ -43,19 +44,40 @@ public IncrementingPollingCounter(string name, EventSource eventSource, Func<dou
         private double _increment;
         private double _prevIncrement;
         private readonly Func<double> _totalValueProvider;
+        private bool _resetStatistics;
+
+        internal void ResetStatistics()
```

Comment:
The name `ResetStatistics` doesn't make sense to me here, it's not actually resetting any statistics. It just causes an extra update on the next timer poll.

Reply:
If we decide to go with this approach I would be happy to find a different name and unify this reset logic for ```IncrementingEventCounter```, ```IncrementingPollingCounter``` and  ```EventCounter```